### DISCUSSION TITLE
PieChart: Support combining smaller slices into an other or rest category

### DISCUSSION THREAD

**Discussion by jdmaloney** [grafana/grafana#38307] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
- Ability to combine values below X%, eg. in v1 pie chart values below say 3% could be combined into a group that has a custom name
- Ability to sort the legend by any of the columns, right now it is just by name column alphabetical

**Why is this needed**:
Feature parity with v1.  Combining small values makes charts more easy to read and understand, sorting also makes it easier to see highest value components.  

I did not see these options as of v8.0.3, but please let me know if I'm somehow missing them.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Reply by taintedkernel** [reply]
I just started exploring the use of pie charts and was looking for the same feature.  Has this been given any consideration as to possibly implement?  Running v8.3.3 here currently.  Thanks!

**Reply by psheehymn** [reply]
We're in the exact same situation. This feature would provide a great deal of benefit for our executive-oriented dashboards.

**Comment by apollo13** [comment]
Using this example from play.grafana.com:
![image](https://user-images.githubusercontent.com/81547/183972444-769493b9-3a83-4721-916e-beeca46e9c72.png)

it would be great if there would be a "other" bucket. After all the database query returned 20 items but only 2 are shown.

**Comment by gitneep** [comment]
Anybody working on this? We have very high cardinality charts that I would love to get into a piechart in a meaningful way, but as it stands this is now impossible. The Piechart becomes invisible when there are more than 100 or so possible keys that want to go in. 

I've tried to limit this in the underlying Infux query, but then the keys have values that peak above the threshold for the influx query, they make it into the piechart but with totally wrong mean() values, as only the timestamps where the value peaks above the threshold are taken into account to calculate the mean(). This is a bug, ALL timestamps for a given period for a given key/value should be used as the total timestamps to be divided by.

Either fix the mean() calculation, or bring back the "Combine" function from the old piechart please.

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/grafana-bi-squad

**Comment by henri9813** [comment]
This issue has been recreated here: https://github.com/grafana/grafana/issues/83689
